### PR TITLE
Task2 release registr experement

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Infrastructure/IMaper.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using MoneySplitter.Models;
+
+
+namespace MoneySplitter.Infrastructure
+{
+    public interface IMaper
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Infrastructure/IMaper.cs]: rename mapper

File: MoneySplitter.Infrastructure/IMaper.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using MoneySplitter.Models;
+
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Infrastructure/IMaper.cs]: remove

File: MoneySplitter.Win10/ViewModels/RegistrViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        {
+            get { return _email; }
+            set
+            {
+                _email = value;
+                NotifyOfPropertyChange(nameof(Email));
+            }
+        }
+
+        public string Password
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/RegistrViewModel.cs]: remove

File: MoneySplitter.Win10/ViewModels/RegistrViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        {
+            get { return _creditCardNumber; }
+            set
+            {
+                _creditCardNumber = value;
+                NotifyOfPropertyChange(nameof(CreditCardNumber));
+            }
+        }
+
+
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/RegistrViewModel.cs]: remove

File: MoneySplitter.Services/API/QueryApiService.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    public class QueryApiService : IQueryApiService
+    {
+        public async Task<TResultQuery> PostQueryAsync<TResultQuery, TBodyQuery>(TBodyQuery bodyQuery, Uri uri)
+            where TResultQuery : class
+            where TBodyQuery : class
+        {
+            TResultQuery resultQuery;
+            using (var httpClient = new HttpClient())
+            {
+
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Services/API/QueryApiService.cs]: remove space

File: MoneySplitter.Services/API/ApiUrlBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using MoneySplitter.Infrastructure;
+
+namespace MoneySplitter.Services.Api
+{
+    public  class ApiUrlBuilder : IApiUrlBuilder
+    {
+        private const string WebApiUrl = "http://moneytransfer.azurewebsites.net";
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Services/API/ApiUrlBuilder.cs]: fix defines

File: MoneySplitter.Infrastructure/IUserModel.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿namespace MoneySplitter.Infrastructure
+{
+    public interface IUserModel
```
Comment (Vovanella95) [inline MoneySplitter.Infrastructure/IUserModel.cs]: Please remove this interface

File: MoneySplitter.Services/API/ApiUrlBuilder.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+                Defines.Api.Session.SESSION,
+                Defines.Api.Session.SIGN_IN));
+        }
+
+        public Uri Registration()
+        {
+            return new Uri(string.Concat(
+                Defines.Api.WEB_API_URL,
+                Defines.Api.Session.SESSION,
+                Defines.Api.Session.REGISTR));
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/ApiUrlBuilder.cs]: registEr, please rename anywhere this word

File: MoneySplitter.Services/API/Defines.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    public static class Defines
+    {
+        public static class Api
+        {
+            public const string WEB_API_URL = "http://moneytransfer.azurewebsites.net";
+            public static class Session
+            {
+                public const string SESSION = "/session";
+                public const string SIGN_IN = "/signin";
+                public const string REGISTR = "/register";
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/Defines.cs]: registEr

File: MoneySplitter.Services/API/SessionApiService.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            {
+                Email = email,
+                Password = password
+            };
+
+            var signInUri = _urlBuilder.Authorization();
+
+            var dataUser = await _queryApiService.PostQueryAsync<DataUser, LoginModel>(loginModel, signInUri);
+
+            return _maper.DataUserToConvertUserModel(dataUser);
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/SessionApiService.cs]: Strange name of method. ConvertUserDataToUserModels sounds better

File: MoneySplitter.Services/DataModels/DataUser.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
+﻿using MoneySplitter.Infrastructure;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace MoneySplitter.Services.DataModels
+{
+    [DataContract]
+    public class DataUser : IUserModel
```
Comment (Vovanella95) [inline MoneySplitter.Services/DataModels/DataUser.cs]: IUserModel - not needed

File: MoneySplitter.Services/Interfases/IMaper.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+using MoneySplitter.Models.Session;
+using MoneySplitter.Services.DataModels;
+
+namespace MoneySplitter.Services.Inerfaces
+{
+    public interface IMapper
+    {
+        UserModel DataUserToConvertUserModel(DataUser dataUser);
+        DataRegisterUser RegisterModelToConverDataRegisterModel(RegisterModel registerModel);
+
```
Comment (Vovanella95) [inline MoneySplitter.Services/Interfases/IMaper.cs]: Space

Comment (Platton96) [reply MoneySplitter.Services/Interfases/IMaper.cs]: remove

File: MoneySplitter.Services/API/Defines.cs (hunk 9 above)
Comment (Platton96) [reply MoneySplitter.Services/API/Defines.cs]: rename

File: MoneySplitter.Services/API/ApiUrlBuilder.cs (hunk 8 above)
Comment (Platton96) [reply MoneySplitter.Services/API/ApiUrlBuilder.cs]: Rename

File: MoneySplitter.Services/API/SessionApiService.cs (hunk 10 above)
Comment (Platton96) [reply MoneySplitter.Services/API/SessionApiService.cs]: Rename

**Pull request merged**